presto-spi/src/main/java/com/facebook/presto/spi/block/BlockUtil.java

```diff
@@ -112,43 +127,65 @@ static Slice compactSlice(Slice slice, int index, int length)
         return Slices.copyOf(slice, index, length);
     }
 
-    static boolean[] compactBooleanArray(boolean[] array, int index, int length)
+    /**
+     * Returns the array starting at specified index and extends for the specified length.
+     * <p>
+     * If the whole array is requested, the input array will be returned.
+     */
+    static boolean[] compactArray(boolean[] array, int index, int length)
     {
         if (index == 0 && length == array.length) {
             return array;
         }
         return Arrays.copyOfRange(array, index, index + length);
     }
 
-    static byte[] compactByteArray(byte[] array, int index, int length)
+    static byte[] compactArray(byte[] array, int index, int length)
     {
         if (index == 0 && length == array.length) {
             return array;
         }
         return Arrays.copyOfRange(array, index, index + length);
     }
 
-    static short[] compactShortArray(short[] array, int index, int length)
+    static short[] compactArray(short[] array, int index, int length)
     {
         if (index == 0 && length == array.length) {
             return array;
         }
         return Arrays.copyOfRange(array, index, index + length);
     }
 
-    static int[] compactIntArray(int[] array, int index, int length)
+    static int[] compactArray(int[] array, int index, int length)
     {
         if (index == 0 && length == array.length) {
             return array;
         }
         return Arrays.copyOfRange(array, index, index + length);
     }
 
-    static long[] compactLongArray(long[] array, int index, int length)
+    static long[] compactArray(long[] array, int index, int length)
     {
         if (index == 0 && length == array.length) {
             return array;
         }
         return Arrays.copyOfRange(array, index, index + length);
     }
+
+    /**
+     * Returns if all corresponding pairs of elements in the two arrays are the same object.
```

Comment:
```
Returns <tt>true</tt> if the two specified arrays contain the same object in every position.
Unlike the {@link Arrays#equals(Object[],Object[])} method, this method compares using reference equals.
```

(I adapted this from `Arrays.deepEquals`. Key take away: always write "returns true if" instead of "returns if/whether")